crates/sui-framework/packages/sui-framework/sources/crypto/group_ops.move

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Generic Move and native functions for group operations.
+module sui::group_ops {
+
+    use std::vector;
+    use sui::bcs;
+
+    friend sui::bls12381;
+
+    #[allow(unused_const)]
+    const ENotSupported: u64 = 0; // Operation is not supported by the network.
+    const EInvalidInput: u64 = 1;
+    #[allow(unused_const)]
+    const EInputTooLong: u64 = 2;
+    const EInvalidBufferLength: u64 = 3;
+
+    /////////////////////////////////////////////////////
+    ////// Generic functions for group operations. //////
+
+    // The caller provides a type identifer that should match the types of enum [Groups] in group_ops.rs.
+
+    // General wrapper for all group elements.
+    struct Element<phantom T> has store, copy, drop {
+        bytes: vector<u8>,
+    }
+
+    public fun bytes<G>(e: &Element<G>): &vector<u8> {
+        &e.bytes
+    }
+    
+    public fun equal<G>(e1: &Element<G>, e2: &Element<G>): bool {
+        &e1.bytes == &e2.bytes
+    }
+
+    // Fails if the bytes are not a valid group element and 'is_trusted' is false.
+    public(friend) fun from_bytes<G>(type: u8, bytes: &vector<u8>, is_trusted: bool): Element<G> {
```

Comment:
@benr-ml do we pad the bytes with leading zeros? Can we have a test around this (for no padding, 1-2 zero bytes padding etc)?

Reply:
we expect a fixed size per type, see in the tests LONG_SCALAR_BYTES and SHORT_SCALAR_BYTES for example